qos/evm/check.go

```diff
@@ -24,11 +27,21 @@ var _ gateway.QoSEndpointCheckGenerator = &EndpointStore{}
 // TODO_IMPROVE(@commoddity): implement QoS check expiry functionality and use protocol.EndpointAddr
 // to filter out checks for any endpoint which has acurrently valid QoS data point.
 func (es *EndpointStore) GetRequiredQualityChecks(_ protocol.EndpointAddr) []gateway.RequestQoSContext {
-	return []gateway.RequestQoSContext{
+	qualityChecks := []gateway.RequestQoSContext{
 		getEndpointCheck(es.logger, es, withChainIDCheck),
 		getEndpointCheck(es.logger, es, withBlockHeightCheck),
-		// TODO_FUTURE: add an archival endpoint check.
 	}
+
+	// If all of the following are true:
+	//  - Service is configured to perform an archival check
+	//  - Has calculated an expected archival balance (balance at a specific block number for a specific address for a particular chain)
+	// Then:
+	// - Add the archival check to the list of qos checks to perform on every hydrator run
+	if es.serviceState.shouldPerformArchivalCheck() {
```

Comment:
WDYT of the following to make the responsibilities more clear without over extending the PR's scope:
- The ServiceState can return a (JSONRPC request, bool) pair, the latter indicating whether an archival check should be added.
- The code here simply updates the ID of the request (if necessary).

Reply:
I have refactored to use this suggestion; I like it.